main.tf

```diff
@@ -127,10 +158,18 @@ resource "aws_db_subnet_group" "db" {
 }
 
 # Create a DNS name for the resource
-resource "aws_route53_record" "dns" {
+resource "aws_route53_record" "dns_including_dbname" {
```

Comment:
You may need to add a count for these two route53 records, similar to what's done above. Otherwise the code will attempt to create 2 records, and one will fail because the `aws_db_instance` doesn't exist.